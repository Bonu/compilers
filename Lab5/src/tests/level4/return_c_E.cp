// : Test for flagging of method decl-type and return-type mismatch.

class Main{

  void void_void() {
    return;
  }

  void void_char() {
    return ('2');
  }

  void void_int() {
    return (21);
  }

  void void_bool() {
    return (true);
  }

  void void_string() {
    return "Hi";
  }

  void void_bool2() {
    return (false);
  }

  void void_error() {
    return (a);   // two error messages--not allowed to return a value 
  }

  char char_void() {
    return;
  }

  char char_char() {
    return ('1');
  }

  char char_int() {
    return (11);
  }

  char char_bool() {
    return (true);
  }

  char char_string() {
    return ("Hi!");
  }

  char char_bool2() {
    return false;
  }

  char char_error() {
    return (a);   // one error message--assume the return value is correct
  }

  int int_void() {
    return;
  }

  int int_char() {
    return ('1');
  }

  int int_int() {
    return (11);
  }

  int int_bool() {
    return (true);
  }

  int int_string() {
    return ("Hi!");
  }

  int int_bool2() {
    return (false);
  }

  int int_error() {
    return (a);   // one error message--assume the return value is correct
  }

  bool bool_void() {
    return;
  }

  bool bool_char() {
    return ('1');
  }

  bool bool_int() {
    return (11);
  }

  bool bool_bool() {
    return (true);
  }

  bool bool_string() {
    return ("Hi!");
  }

  bool bool_error() {
    return (a);   // one error message--assume the return value is correct
  }

  bool bool_bool2() {
    return false;
  }

  int k1(int z) {
    return (la[z]);
  }
  int i;
  int j;
  float la[50];
  string sa[50];
  char ca[50];

  int k2(int z) {
    return (i+j+la[z]);
  }

  int k3(int z) {
    return (i+j+sa[z]);  // must not give invalid return type error message
  }

  int k4(int z) {
    return (sa[z]);
  }

  char k5(int z) {
    return (i+j+la[z]);
  }

  char k6(int z) {
    return (sa[z]);
  }

}

