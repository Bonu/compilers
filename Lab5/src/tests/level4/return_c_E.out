line 10: cannot return a value from methods with a void result type
line 14: cannot return a value from methods with a void result type
line 18: cannot return a value from methods with a void result type
line 22: cannot return a value from methods with a void result type
line 26: cannot return a value from methods with a void result type
line 30: unable to resolve name -- a
line 30: cannot return a value from methods with a void result type
line 34: a return expression is required in methods with a return type
line 42: the return value does not match the return type of method 'char_int'
          -- expects type `char' but found an expression with type `int'
line 46: the return value does not match the return type of method 'char_bool'
          -- expects type `char' but found an expression with type `bool'
line 50: the return value does not match the return type of method 'char_string'
          -- expects type `char' but found an expression with type `string'
line 54: the return value does not match the return type of method 'char_bool2'
          -- expects type `char' but found an expression with type `bool'
line 58: unable to resolve name -- a
line 62: a return expression is required in methods with a return type
line 74: the return value does not match the return type of method 'int_bool'
          -- expects type `int' but found an expression with type `bool'
line 78: the return value does not match the return type of method 'int_string'
          -- expects type `int' but found an expression with type `string'
line 82: the return value does not match the return type of method 'int_bool2'
          -- expects type `int' but found an expression with type `bool'
line 86: unable to resolve name -- a
line 90: a return expression is required in methods with a return type
line 94: the return value does not match the return type of method 'bool_char'
          -- expects type `bool' but found an expression with type `char'
line 98: the return value does not match the return type of method 'bool_int'
          -- expects type `bool' but found an expression with type `int'
line 106: the return value does not match the return type of method 'bool_string'
          -- expects type `bool' but found an expression with type `string'
line 110: unable to resolve name -- a
line 118: the return value does not match the return type of method 'k1'
          -- expects type `int' but found an expression with type `float'
line 127: the return value does not match the return type of method 'k2'
          -- expects type `int' but found an expression with type `float'
line 131: invalid argument 2 ` sa [ z ] ' for arithmetic operator `+'
          -- expects a numeric argument but found string
line 135: the return value does not match the return type of method 'k4'
          -- expects type `int' but found an expression with type `string'
line 139: the return value does not match the return type of method 'k5'
          -- expects type `char' but found an expression with type `float'
line 143: the return value does not match the return type of method 'k6'
          -- expects type `char' but found an expression with type `string'
line 0: Method `Main.main' has not been defined
30 errors were detected
